discord/ext/commands/cooldowns.py

```diff
@@ -48,6 +50,8 @@
     'MaxConcurrency',
 )
 
+T = TypeVar('T')
```

Comment:
This feels like it needs a bound, though I can't test right now whether it'd work as I expected it to with it.

Reply:
What type would it be bound to? As written, it allows specifying any type to be used, as opposed to strictly limiting to `Message` and `Context`.

Reply:
Yes, and that's the problem

Reply:
I don't see a problem in that. If it were bound and someone wanted to use it to ratelimit reactions or interactions, they'd need to `# type: ignore` the assignment.